glean-core/ffi/src/lib.rs

```diff
@@ -342,27 +342,50 @@ pub extern "C" fn glean_is_first_run() -> u8 {
     with_glean_value(|glean| glean.is_first_run())
 }
 
+// Unfortunately, the way we use CFFI in Python ("out-of-line", "ABI mode") does not
+// allow return values to be `union`s, so we need to use an output parameter instead of
+// a return value to get the task.
```

Comment:
Let's document here that we write the output into `result`.
And also call out that we take the data back in the below function to free it.